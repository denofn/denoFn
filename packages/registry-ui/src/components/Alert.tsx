import React from "react";
import { css } from "otion";

export type AlertProps = {
  title: string;
  message: string;
};

export function Alert({ title, message }: AlertProps) {
  return (
    <div
      className={css({
        paddingLeft: "1rem",
        paddingRight: "1rem",
        color: "rgb(197,48,48)",
        borderStyle: "solid",
        borderColor: "rgb(252,129,129)",
        borderRadius: ".25rem",
        borderWidth: "1px",
        paddingTop: ".75rem",
        paddingBottom: ".75rem",
        position: "relative",
        backgroundColor: "rgb(255,245,245)",
      })}
      role="alert"
    >
      <strong
        className={css({
          fontFamily: "'Rubik', sans-serif",
          fontWeight: "bold",
          display: "block",
          lineHeight: "1.5rem",
        })}
      >
        {title}
      </strong>
      <span>{message}</span>
    </div>
  );
}
