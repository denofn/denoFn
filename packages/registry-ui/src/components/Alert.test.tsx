import { render, cleanup } from "@testing-library/react";
import React from "react";

import { Alert } from "./Alert";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<Alert title="title" message="message" />);
  expect(asFragment()).toMatchSnapshot();
});
