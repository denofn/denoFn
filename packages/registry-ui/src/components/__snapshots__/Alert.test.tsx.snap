// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="_1htmj9j _1m7j13x _dizro5 _fxc8cu _1m1v6p5 _sy4y6p _my9kfw _46ipn5 _1wfxxmo _bjn8wh _3s73u6"
    role="alert"
  >
    <strong
      class="_1k2w1pl _1efi8gv _4zleql _1dw7ik1"
    >
      title
    </strong>
    <span>
      message
    </span>
  </div>
</DocumentFragment>
`;
